artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -689,7 +689,10 @@ public QueueBinding updateQueue(QueueConfiguration queueConfiguration) throws Ex
                queue.setDelayBeforeDispatch(queueConfiguration.getDelayBeforeDispatch().longValue());
             }
             Filter filter = FilterImpl.createFilter(queueConfiguration.getFilterString());
-            if (filter != null && !filter.equals(queue.getFilter())) {
+            if ((filter == null) && (queue.getFilter() != null)) {
```

Comment:
Should think about applying the same to all, e.g. if now null set back to default for address setting (or static default)

Reply:
I've added the config reload test.

I'm not sure about what you mean by the default behavior. Shall I add some sort of constant (like `DEFAULT_FILTER`) or just document the fact that there's no filter by default and that setting it to `null` (by removing it from the configuration) or to `""` reverts it back to the default value?

Reply:
I was meaning other queue configurations that you can update e.g like the one immediately above delaybeforedispatch. E.g. if change of behaviour from currently if null do not update, now youre proposing to make it if null set back to default which for filter is null. We would want that behaviour consistent. Else we end up with some ignoring and some reverting. 

Its very important behaviour is consistent for all, so either all do nothing when new value is null (ignore) or all update back to their defaults (being either the default at address setting or where a default constant its used)

Reply:
I see. I'll take look at that.

Reply:
Hm, I must have overlooked the fact that the other properties behave the same too.
Is the behavior I propose a correct one? Or is there a reason for ignoring null values?

Now that it's obvious - even to me :-) - that the issue is a lot broader, shall I continue with implementing the rest here or would that be a whole new pull request?

My use case in an automatic cluster configuration deployment and the current behavior sort of breaks the idea.

Reply:
Clearing the value using `""` does not work, at least not for a filter.
```
//...
public static Filter createFilter(final SimpleString filterStr) throws ActiveMQException {
      if (filterStr == null || filterStr.length() == 0) {
         return null;
      }
// ...
```
So there would have to be a breaking change anyway, just to achieve this.

I understand the issue with resetting the whole configuration. Maybe there could be a toggle to enable this.

I'm working for a cloud-based EDI provider and our ultimate goal is to have all of our infrastructure configurable via Ansible (or similar tools) so that in case of a major failure there'd be no need to perform too many manual steps in order to rebuild the service from scratch. That makes the Ansible repository the only place to store our configuration, and therefore it would be highly beneficial if I could redeploy the Artemis configuration from there. :-) It would also open the way to create a bit more user friendly tool to design the MQ related infrastructure.
I'd be happy to hear about the way you do it in your company. :-)